File: idioms/javascript/Closure1.js
#! /usr/bin/env node
"use strict";

var context = 'method1 context';
function Method1(){
  return context;
}

function CreateMethod1() {
  var context = 'method1 context?';
  return Method1;
}

function CreateMethod2() {
    var context = 'method2 context';
    function Method2() {
      return context;
    }

    return Method2;
}

var method1 = CreateMethod1();
console.log(method1());

var method2 = CreateMethod2();
console.log(method2());

File: idioms/python/Closure1.py
#! /usr/bin/env python

context = 'method1 context'
def Method1():
    return context;

def CreateMethod1():
    context = 'method1 context?'
    return Method1

def CreateMethod2():
    context = 'method2 context'
    def Method2():
        return context
    return Method2

method1 = CreateMethod1()
print(method1())

method2 = CreateMethod2()
print(method2())
